samples/project-geometry/src/main/java/com/esri/arcgismaps/sample/projectgeometry/components/ProjectGeometryViewModel.kt

```diff
@@ -0,0 +1,116 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.projectgeometry.components
+
+import android.app.Application
+import android.graphics.BitmapFactory
+import android.graphics.drawable.BitmapDrawable
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.geometry.GeometryEngine
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.symbology.PictureMarkerSymbol
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.SingleTapConfirmedEvent
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.projectgeometry.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import java.util.Locale
+
+class ProjectGeometryViewModel(val app: Application) : AndroidViewModel(app) {
+    // create a map with a navigation night basemap style
+    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISStreetsNight).apply {
+        // set the default viewpoint to Redlands,CA
+        initialViewpoint = Viewpoint(34.058, -117.195, 5e4)
+    }
+    // create a MapViewProxy to interact with the MapView
+    val mapViewProxy = MapViewProxy()
+
+    // Create a message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    // setup the red pin marker image as bitmap drawable
+    private val markerDrawable: BitmapDrawable by lazy {
+        // load the bitmap from resources and create a drawable
+        val bitmap = BitmapFactory.decodeResource(app.resources, R.drawable.pin_symbol)
+        BitmapDrawable(app.resources, bitmap)
+    }
+
+    // setup the red pin marker as a Graphic
+    private val markerGraphic: Graphic by lazy {
+        // creates a symbol from the marker drawable
+        val markerSymbol = PictureMarkerSymbol.createWithImage(markerDrawable).apply {
+            // resize the symbol into a smaller size
+            width = 30f
+            height = 30f
+            // offset in +y axis so the marker spawned is right on the touch point
+            offsetY = 25f
+        }
+        // create the graphic from the symbol
+        Graphic(symbol = markerSymbol)
+    }
+
+    val graphicsOverlay = GraphicsOverlay().apply {
+        graphics.add(markerGraphic)
+    }
+
+    private val _infoText = MutableStateFlow(app.resources.getString(R.string.tap_to_begin))
+    val infoText: StateFlow<String> = _infoText
```

Comment:
I don't think we should be formatting the text in the viewmodel if we can do it in the main screen - and we can. We should move text formatting concerns to the screen, and instead expose the point and its projection from the view model.

```suggestion
    // state flow of a point and its projection for presentation in UI
    private val _pointProjectionFlow = MutableStateFlow<PointProjection?>(null)
    val pointFlow = _pointProjectionFlow.asStateFlow()
```

This uses a very simple data class:

`data class PointProjection (val original: Point, val projection: Point)`